packages/vite/src/node/plugins/css.ts

```diff
@@ -622,6 +622,16 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
       return null
     },
 
+    augmentChunkHash(chunk) {
+      if (chunk.viteMetadata.importedCss.size) {
+        let hash = ''
+        for (const id of chunk.viteMetadata.importedCss) {
+          hash += id
+        }
+        return hash
+      }
+    },
```

Comment:
Is this a problem in the case of CSS because we are emitting the CSS asset on `generateBundle`, no? I missed this one, thanks a lot for taking care of it 🙏🏼 
For other emitted assets, we are doing them in transform or renderChunk, so the new rollup hashing algorithm takes care of them IIUC.

Reply:
I think the problem is the [emitFile in renderChunk](https://github.com/vitejs/vite/blob/568a0146b46f5cbaa52b3509f757b23364471197/packages/vite/src/node/plugins/css.ts#L563-L567). We emit the styles but the chunk itself never imports it in code, so the `code` doesn't change hence the `hash` stays the same. So `augmentChunkHash` adds this extra metadata without needing to alter `code`.

The CSS emit in `generateBundle` seems to be for `cssCodeSplit: false` cases only, and since it's a separate CSS file tied to a HTML, the JS chunk hashes don't need to inherit and cascade from the CSS change.

Reply:
Ya. The problem is that later we end up modifying the `code` in `generateBundle` [here](https://github.com/vitejs/vite/blob/3d346c032e070514482ca4f977a244805481d309/packages/vite/src/node/plugins/importAnalysisBuild.ts#L568). Thanks again for this fix!